### PR TITLE
Fix querystate key not found and add tests for function state

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

When a user queries the state of function for a key that doesn't exist, a status code of 500 (INTERNAL SERVER ERROR) is returned.  This is incorrect.  Should return 404

Fix this issue as well as add some tests for function state

### CODE REVIEW

File: pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsemble.java [hunk 1]
```diff
@@ -373,6 +373,13 @@ public void start() throws Exception {
         runZookeeper(1000);
         initializeZookeper();
         runBookies(conf);
+
+        if (enableStreamStorage) {
+            runStreamStorage(new CompositeConfiguration());
+        }    }
```
Comment (sijie) [inline pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsemble.java]:
```suggestion
            }    
        }
```

**Comment by jerrypeng** [comment]
rerun cpp tests

**Comment by srkukarni** [comment]
rerun cpp tests

**Comment by jerrypeng** [comment]
rerun cpp tests

**Comment by jerrypeng** [comment]
rerun cpp tests

**Pull request merged**